graph-manager-docs/source/setup-analytics.md

```diff
@@ -107,29 +91,53 @@ query AuthorForPost {
+For a reference implementation, see the `sendReport()` method in the [TypeScript reference agent](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-engine-reporting/src/agent.ts#L210).
 
-Reporting agents can authenticate either via the `X-Api-Key` header or the `authtoken` cookie with the service API key.
+#### Tuning reporting behavior
 
-We recommend implementing retries with backoff on 5xx responses and network errors and allowing for the batching size to be tunable by the user. Additionally, we recommend adding a shutdown hook to send all pending reports to ensure that healthy server shutdowns do not result in missing data, as this tracing information is especially important.
+We recommend implementing retries with backoff when you encounter `5xx` responses or networking errors when communicating with the reporting endpoint. Additionally, implement a shutdown hook to make sure you push all pending reports before your server initiates a healthy shutdown.
 
-> NOTE: In the future, we plan to release a local aggregation and sampling agent that could be used to lessen the bandwidth requirements on reporting agents.
+### Implementing additional reporting features
 
-### 4. [Optional] Advanced Reporting Features
+The reference TypeScript implementation includes several features that you might want to include in your implementation. All of these features are implemented in the agent itself, and are documented in the interface description for the `EngineReportingOptions` of [the agent](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-engine-reporting/src/agent.ts#L48).
 
-The reference TypeScript implementation also includes several more advanced features which may be worth porting to new implementations. All of these features are implemented in the agent itself and are documented in the interface description for the EngineReportingOptions of [the agent](https://github.com/apollographql/apollo-server/blob/master/packages/apollo-engine-reporting/src/agent.ts#L51).
+For example, the option to send reports immediately is particularly useful for GraphQL servers that run in a serverless environment, like AWS Lambda or Google Cloud Functions.
 
-For example, the option to send reports immediately may be particularly useful to GraphQL servers running in a serverless environment, like AWS Lambda or Google Cloud Functions.
+Another important feature is the ability to restrict which information is sent to Graph Manager, particularly to avoid reporting personal data. Because personal data most commonly appears in variables and headers, the TypeScript agent offers options for `privateVariables` and `privateHeaders`.
```

Comment:
those options are deprecated and replaced with `sendVariablesValues` and `sendHeaders` now (and oops I rewrote a line like this above)